shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeExecutorServiceImpl.java

```diff
@@ -95,7 +97,7 @@ public Deployment fetchOrCreateBridgeExecutorDeployment(BridgeExecutor bridgeExe
         try {
             environmentVariables.add(new EnvVarBuilder().withName(Constants.BRIDGE_EXECUTOR_PROCESSOR_DEFINITION_ENV_VAR).withValue(objectMapper.writeValueAsString(bridgeExecutor.toDTO())).build());
         } catch (JsonProcessingException e) {
-            LOGGER.error("Could not serialize Processor Definition while setting executor deployment environment variables", e);
+            mdc(LOGGER, bridgeExecutor).error("Could not serialize Processor Definition while setting executor deployment environment variables", e);
```

Comment:
This is the unavoidable MDC setup for the Shard.

I wrap the boilder plate in a `mdc` static helper method... it's the best I can come up with at the moment.

Reply:
@manstis I don't know is this is necessarily better than what you're proposing, but an alternative could be something like the following:

```
class JsonLoggingProvider implements JsonProvider {
 /*
        Things that provide additional fields to be Json logged can implement this
     */
    private interface JsonLoggable {
        Map<String, String> getJsonLogValues();
    }

    @Override
    public void writeTo(JsonGenerator generator, ExtLogRecord event) throws IOException {
        Map<String, String> jsonParams = resolveJsonParams(event.getParameters());
        for (Map.Entry<String, String> e : jsonParams.entrySet()) {
            generator.writeStringField(e.getKey(), e.getValue());
        }
    }

    Map<String, String> resolveJsonParams(Object[] parameters) {
        Map<String, String> jsonParams = new HashMap<>();
        Arrays.stream(parameters).filter(o -> o instanceof JsonLoggable).forEach(o -> jsonParams.putAll(((JsonLoggable) o).getJsonLogValues()));
        return jsonParams;
    }
}
```

So in ^ model, any entities that want to contribute to the top level logging on the JSON log statement, implement the `JsonLoggable` interface to return a set of key/value that should be combined into the JSON log object. The `JsonProvider` then looks at the log parameters for each logging call and simply adds them if the entity implements that interface.

Benefits I can see to this approach are:

1. Log implementation is a bit more centralised again within the `JsonProvider`
2. It's transparent to the end developer. I don't need to remember to call the `mdc` wrapper.

p.s My implementation here is probably less than optimal, just more for example/discussion

Reply:
@robpblake I've cleared up logging of `workerId` to just the `manager`. I've also removed the static `mdc(..)` wrapper and related code and elected for an approach analogous to your proposal but limited to `customerId`. I have made DTOs that contain `customerId` implement `CustomerIdProvider` and, if a parameter passed to the `JsonProvider` class implements it extract the `customerId`. 

It has the down-side that we must pass the full DTO to `LOGGER` and not, for example, simply `BridgeDto.getId()`. This limitation is restricted to the `shard-operator` only - as `customerId` needs to be extracted from DTOs and is not available at a more generic _global_ level (e.g. JWT token, `application.properties` etc).

Have a look and let me now if it's better (than it was).